src/front/wgsl/mod.rs

```diff
@@ -258,21 +227,13 @@ impl<'a> Error<'a> {
                 labels: vec![(bad_span.clone(), "unexpected components".into())],
                 notes: vec![],
             },
-            Error::BadU32(ref bad_span, ref err) => ParseError {
-                message: format!(
-                    "expected unsigned integer literal, found `{}`",
-                    &source[bad_span.clone()],
-                ),
-                labels: vec![(bad_span.clone(), "expected unsigned integer".into())],
-                notes: vec![err.to_string()],
-            },
-            Error::BadI32(ref bad_span, ref err) => ParseError {
+            Error::BadNumber(ref bad_span, ref err) => ParseError {
                 message: format!(
-                    "expected integer literal, found `{}`",
-                    &source[bad_span.clone()],
+                    "`{}` {}",
```

Comment:
I think this is going to produce error messages like:

> \`10.0h\` unimplemented f16 type

which is a little odd. It seems like

> unimplemented f16 type: \`10.0h\`

would be better.